x/gamm/pool-models/balancer/amm_test.go

```diff
@@ -445,6 +445,234 @@ func TestCalcJoinPoolShares(t *testing.T) {
 	}
 }
 
+// TestGetPoolAssetsByDenom tests if `GetPoolAssetsByDenom` succesfully creates a map of denom to pool asset
+// given pool asset as parameter
+func TestGetPoolAssetsByDenom(t *testing.T) {
+	testCases := []struct {
+		name                      string
+		poolAssets                []balancer.PoolAsset
+		expectedPoolAssetsByDenom map[string]balancer.PoolAsset
```

Comment:
This field seems redundant?

Reply:
Why would you think so?

`poolAssets` is the input to the function we are testing, `expectedPoolAssetsByDenom` is the expected output.

and we are, essentially, testing the conversion from `[]balancer.PoolAsset` to `map[string]balancer.PoolAsset`

Reply:
Moved this logic to: https://github.com/osmosis-labs/osmosis/pull/1738